### PR TITLE
cdrdao: patch only on mac, this breaks the build on Linux

### PR DESCRIPTION
**Pull request by iMichka**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/cdrdao.rb [hunk 1]
```diff
@@ -21,11 +21,13 @@ class Cdrdao < Formula
...
   depends_on "mad"
 
-  # first patch fixes build problems under 10.6
-  # see https://sourceforge.net/p/cdrdao/patches/23/
-  patch do
-    url "https://sourceforge.net/p/cdrdao/patches/_discuss/thread/205354b0/141e/attachment/cdrdao-mac.patch"
-    sha256 "ee1702dfd9156ebb69f5d84dcab04197e11433dd823e80923fd497812041179e"
+  on_macos do
+    # first patch fixes build problems under 10.6
+    # see https://sourceforge.net/p/cdrdao/patches/23/
```
Comment (SMillerDev) [inline Formula/cdrdao.rb:26]: Do those issues still exist 10 major macOS versions later?

**Comment by Bo98** [comment]
Somewhat concerning that a `ScsiIf-osx.cc` patch would affect the Linux build. Shouldn't be compiling that file at all.

File: Formula/cdrdao.rb (hunk 1 above)
Comment (iMichka) [reply Formula/cdrdao.rb:26]: Yes, I tested without the patch on my Mac:

```
In file included from ScsiIf-osx.cc:22:
./decodeSense.cc:86:3: error: use of undeclared identifier 'log_message'
  log_message(-2, "SCSI command failed:");
  ^
./decodeSense.cc:87:3: error: use of undeclared identifier 'log_message'
  log_message(-2, "  sense key 0x%x: %s.", code,
  ^
./decodeSense.cc:91:5: error: use of undeclared identifier 'log_message'
    log_message(-2, "  additional sense code: 0x%x", buf[0x0c]);
    ^
./decodeSense.cc:94:5: error: use of undeclared identifier 'log_message'
    log_message(-2, "  additional sense code qualifier: 0x%x", buf[0x0d]);
    ^
4 errors generated.
```

**Comment by iMichka** [comment]
You are right, I think this has been disabled by mistake a long time ago: https://github.com/Linuxbrew/legacy-linuxbrew/pull/880

I tested and it builds fine with the patch on Linux. Closing here, I'll remove that workaround in linuxbrew-core.

**Pull request closed without merging**